source/acceptance_test.go

```diff
@@ -0,0 +1,219 @@
+)
+
+var (
+	startTime time.Time
+)
+
+func TestAcceptance(t *testing.T) {
+	client, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+	sdk.AcceptanceTest(t, AcceptanceTestDriver{
+		Config: AcceptanceSourceTestDriverConfig{
+			Connector: sdk.Connector{
+				NewSpecification: marketo.Specification,
+				NewSource:        source.NewSource,
+			},
+			SourceConfig: getConfigs(),
+			GoleakOptions: []goleak.Option{
+				goleak.IgnoreCurrent(),
+				goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+				// external dependency - minimarketo
+				goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+				goleak.IgnoreTopFunction("net/http.(*persistConn).roundTrip"),
+				goleak.IgnoreTopFunction("net/http.setRequestCancel.func4"),
+			},
+			BeforeTest: func(t *testing.T) {
+				iterator.InitialDate = time.Now().UTC()
+				startTime = time.Now().UTC()
+			},
+			AfterTest: func(t *testing.T) {
+				t.Cleanup(func() {
+					err := cleanUp(client, startTime)
+					if err != nil {
+						t.Error(err)
+					}
+				})
+			},
+		},
+	},
+	)
+}
+
+// AcceptanceTestDriver implements sdk.AcceptanceTestDriver
+type AcceptanceTestDriver struct {
+	Config AcceptanceSourceTestDriverConfig
+	sdk.ConfigurableAcceptanceTestDriver
+}
+
+// AcceptanceSourceTestDriverConfig contains the configuration for
+// AcceptanceTestDriver.
+type AcceptanceSourceTestDriverConfig struct {
+	// Connector is the connector to be tested.
+	Connector sdk.Connector
+
+	// SourceConfig should be a valid config for a source connector, reading
+	// from the same location as the destination will write to.
+	SourceConfig map[string]string
+	// DestinationConfig should be a valid config for a destination connector,
+	// writing to the same location as the source will read from.
+	DestinationConfig map[string]string
+
+	// BeforeTest is executed before each acceptance test.
+	BeforeTest func(t *testing.T)
+	// AfterTest is executed after each acceptance test.
+	AfterTest func(t *testing.T)
+
+	// GoleakOptions will be applied to goleak.VerifyNone. Can be used to
+	// suppress false positive goroutine leaks.
+	GoleakOptions []goleak.Option
+}
+
+func (d AcceptanceTestDriver) DestinationConfig(*testing.T) map[string]string {
+	return map[string]string{}
+}
+func (d AcceptanceTestDriver) Connector() sdk.Connector {
+	return d.Config.Connector
+}
+
+func (d AcceptanceTestDriver) SourceConfig(*testing.T) map[string]string {
+	return d.Config.SourceConfig
+}
+
+func (d AcceptanceTestDriver) BeforeTest(t *testing.T) {
+	// before test check if the test should be skipped
+	d.Skip(t)
+
+	if d.Config.BeforeTest != nil {
+		d.Config.BeforeTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) AfterTest(t *testing.T) {
+	if d.Config.AfterTest != nil {
+		d.Config.AfterTest(t)
+	}
+}
+
+func (d AcceptanceTestDriver) GoleakOptions(_ *testing.T) []goleak.Option {
+	return d.Config.GoleakOptions
+}
+
+func (d AcceptanceTestDriver) ReadTimeout() time.Duration {
+	return time.Minute * 5
+}
```

Comment:
It looks like we don't need these methods anymore.

Reply:
when i delete these methods tests are failing

Reply:
@rustiever because of timeouts? Did you update to the latest SDK?

Reply:
yes